src/libs/OptimisticReportNames.ts

```diff
@@ -290,6 +313,25 @@ function updateOptimisticReportNamesFromUpdates(updates: OnyxUpdate[], context:
         updatesCount: updates.length,
     });
 
+    /**
+     * Helper function to create a report name update with pending field if needed
+     */
+    function createReportNameUpdate(reportID: string, reportNameUpdate: ReportNameUpdate): OnyxUpdate {
+        // @ts-expect-error - will be solved in https://github.com/Expensify/App/issues/73830
+        return {
+            key: getReportKey(reportID),
+            onyxMethod: Onyx.METHOD.MERGE,
+            value: {
+                reportName: reportNameUpdate.name,
+                ...(reportNameUpdate.isPending && {
+                    pendingFields: {
+                        reportName: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                    },
+                }),
```

Comment:
I’m not sure about this change. The normal way would be to add `pendingFields` in the optimistic data and remove it in the failure and success data while making the API request.

@mohammadjafarinejad, would it be possible to add `pendingFields` while making API requests like requestMoney, deleteMoney, update, etc., only if we use custom report fields?

Reply:
@mohammadjafarinejad We used to remove the pendingFields by setting them to null in the success and failure data. How should we handle it in this case?

Reply:
I think we should bring in someone who has worked on this project for some suggestions.

cc: @arosiclair